core/trace/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -51,11 +51,21 @@ trait Tracer[F[_]] extends TracerMacro[F] {
     */
   def currentSpanContext: F[Option[SpanContext]]
 
-  /** Returns the current span if one exists in the local scope, or a no-op span
-    * otherwise.
+  /** @return
+    *   the current span if one exists in the local scope, or a no-op span
+    *   otherwise
     */
   def currentSpanOrNoop: F[Span[F]]
 
+  /** @return
+    *   the current span if one exists in the local scope (even if it's a no-op
+    *   span), or raises an error otherwise
+    *
+    * @throws java.lang.IllegalStateException
+    *   when called while not inside a span, indicating programmer error
+    */
+  def currentSpanOrThrow(implicit F: MonadCancelThrow[F]): F[Span[F]]
```

Comment:
`ApplicativeThrow` should be enough.

```suggestion
  def currentSpanOrThrow(implicit F: MonadCancelThrow[F]): F[Span[F]]
  def currentSpanOrThrow: F[Span[F]]
```

We should provide `ApplicativeThrow` via the constructor in the same way, we provide `Applicative.`